pkg/cfn/components/lambdafunction/lambdafunction.go

```diff
@@ -96,9 +101,56 @@ func NewRotateLambda(
 		Key:          key,
 		Role:         role,
 		Env: map[string]string{
-			"SECRETS_MANAGER_ENDPOINT": cloudformation.Sub("https://secretsmanager.${AWS::Region}.amazonaws.com"),
+			"SECRETS_MANAGER_ENDPOINT": cloudformation.Select(
+				0,
+				[]string{cloudformation.GetAtt(secretsManagerVPCEndpoint.Name(), "DnsEntries")},
+			),
 		},
 		SecurityGroupID: securityGroup,
 		SubnetIds:       subnetIDs,
 	})
 }
+
+func PatchRotateLambda(lambdaFunctionName, secretsManagerVPCEndpointName string, template []byte) ([]byte, error) {
+	patchJSON := []byte(fmt.Sprintf(`[
```

Comment:
This would probably be cleaner as a marshalled struct. (btw, we have the structs available in pkg/scaffold/resources/patch.go

Reply:
I guess you are talking about the `op, path, value` part only?

Reply:
Yeapp